angular/projects/admin/src/app/admin/settings/settings-dialog-manage-languages/settings-dialog-manage-languages.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit, Inject } from '@angular/core';
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
+
+@Component({
+  selector: 'tanam-settings-dialog-manage-languages',
+  templateUrl: './settings-dialog-manage-languages.component.html',
+  styleUrls: ['./settings-dialog-manage-languages.component.scss']
+})
+export class SettingsDialogManageLanguagesComponent implements OnInit {
+
+  readonly languagesForm: FormGroup = this.formBuilder.group({
+    languages: this.formBuilder.array([], [Validators.required]),
+  });
+
+  constructor(
+    @Inject(MAT_DIALOG_DATA) public data: any,
+    private dialogRef: MatDialogRef<SettingsDialogManageLanguagesComponent >,
+    private readonly formBuilder: FormBuilder,
+
+  ) { }
+
+
+  ngOnInit() {
+    for (const language of this.data) {
+      this.addLanguage(language);
+    }
+  }
+
+  get languagesFormArray() {
+    return this.languagesForm.get('languages') as FormArray;
+  }
+
+  addLanguage(language?: string) {
+    console.log(`[SettingsLanguageComponent:addLanguage] ${JSON.stringify(language)}`);
+    const newLanguageLine = this.formBuilder.group({
+      name: [
+        language, [
+          Validators.required,
+        ],
+      ],
+    });
+
+    this.languagesFormArray.push(newLanguageLine);
+  }
+
+  removeLanguage(index: number) {
+    this.languagesFormArray.removeAt(index);
+  }
+
+  save() {
+    const formData = this.languagesForm.value;
+    const languages = {
+      languages: formData.languages.map((language: any) => language['name']),
```

Comment:
Use property name `code` or `id` for the langauge code ("id", "en", ...)

```typescript
interface LanguageOption {
  id: string; // en, id, se...
  title: string; // English title
  nativeTitle: string; // Untranslated title of the language in its own local alphabet and language
}
```

Reply:
Solved on this commit https://github.com/oddbit/tanam/pull/51/commits/c25615f417113730c9b6179b0b828fbe56e50eda